paddle/fluid/framework/tensor_test.cc

```diff
@@ -116,7 +116,7 @@ TEST(Tensor, MutableData) {
     EXPECT_NE(p1, nullptr);
     // set src_tensor a new dim with large size
     // momery is supposed to be re-allocated
-    p2 = src_tensor.mutable_data<float>(framework::make_ddim({3, 4}),
+    p2 = src_tensor.mutable_data<float>(framework::make_ddim({3, 1024}),
```

Comment:
Because `allocation->size()` may be larger than `requested_size` in the new design. For example, `allocation->size()` is always aligned to 256 in GPU. Therefore, `allocation->size()` would be larger than `requested_size` in most cases. In this way, this unittest would fail.

I have added some notes about it in the description of this PR.